POI.Core/Models/BeatSavior/SongData.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using NodaTime;
+using POI.Core.Models.BeatSavior.Trackers;
+
+namespace POI.Core.Models.BeatSavior
+{
+	public class SongData
+	{
+		[JsonPropertyName("_id")]
+		public string Id { get; init; } = string.Empty;
+
+		[JsonPropertyName("songDataType")]
+		public int SongDataType { get; init; }
+
+		[JsonPropertyName("playerID")]
+		public string PlayerId { get; init; } = string.Empty;
+
+		[JsonPropertyName("songID")]
+		public string SongId { get; init; } = string.Empty;
+
+		[JsonPropertyName("songDifficulty")]
+		public string SongDifficulty { get; init; } = string.Empty;
+
+		[JsonPropertyName("songName")]
+		public string SongName { get; init; } = string.Empty;
+
+		[JsonPropertyName("songArtist")]
+		public string SongArtist { get; init; } = string.Empty;
+
+		[JsonPropertyName("songMapper")]
+		public string SongMapper { get; init; } = string.Empty;
+
+		[JsonPropertyName("gameMode")]
+		public string GameMode { get; init; } = string.Empty;
+
+		[JsonPropertyName("songDifficultyRank")]
+		public int SongDifficultyRank { get; init; }
+
+		[JsonPropertyName("songSpeed")]
+		public double SongSpeed { get; init; }
+
+		[JsonPropertyName("songStartTime")]
+		public double SongStartTime { get; init; }
+
+		[JsonPropertyName("songDuration")]
+		public double SongDuration { get; init; }
+
+		[JsonPropertyName("trackers")]
+		public SongTrackers Trackers { get; init; } = null!;
+
+		[JsonPropertyName("deepTrackers")]
+		public DeepTrackers DeepTrackers { get; set; } = null!;
```

Comment:
DeepTrackers should be nullable, also init-only setter
```suggestion
		public DeepTrackers? DeepTrackers { get; init; }
```